OWML.ModHelper/ModMergedConfig.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+using OWML.Common;
+using OWML.Logging;
+
+namespace OWML.ModHelper
+{
+    public class ModMergedConfig : IModConfig
+    {
+        public bool Enabled { get => _userConfig.Enabled; set => _userConfig.Enabled = value; }
+        public Dictionary<string, object> Settings
+        {
+            get {
+                var settings = new Dictionary<string, object>(_defaultConfig.Settings);
+                _userConfig.Settings.ToList().ForEach(x => settings[x.Key] = x.Value);
+                return settings;
+            }
+            set => _userConfig.Settings = value;
+        }
+
+        private IModConfig _userConfig;
+        private readonly IModConfig _defaultConfig;
+        private readonly IModManifest _manifest;
+
+        public ModMergedConfig(IModConfig userConfig, IModConfig defaultConfig, IModManifest manifest)
+        {
+            _userConfig = userConfig;
+            _defaultConfig = defaultConfig;
+            _manifest = manifest;
+            FixConfigs();
+        }
+
+        public T GetSettingsValue<T>(string key)
+        {
+            if (_userConfig.Settings.ContainsKey(key))
+            {
+                return _userConfig.GetSettingsValue<T>(key);
+            }
+            return _defaultConfig.GetSettingsValue<T>(key);
```

Comment:
i *think* you can do 
`return `_userConfig.GetSettingsValue<T>(key) ?? _defaultConfig.GetSettingsValue<T>(key);`

Reply:
I started off that way, but get settings value isn't prepared for missing keys.

Reply:
Ah, then please prepare it for that ;)

Reply:
Return default value of the relevant type, in that case

Reply:
Ah yeah, that was the problem. It returns the default value of the type, which I don't want. So I need to manually check the dictionary.